cloudamqp/resource_cloudamqp_alarm.go

```diff
@@ -100,15 +100,36 @@ func resourceAlarmCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	log.Printf("[DEBUG] cloudamqp::resource::alarm::create params: %v", params)
 
-	data, err := api.CreateAlarm(d.Get("instance_id").(int), params)
-	log.Printf("[DEBUG] cloudamqp::resource::alarm::create data: %v", data)
+	if d.Get("type") == "notice" {
+		log.Printf("[DEBUG] cloudamqp::resource::alarm::create type is 'notice', skipping creation")
+		log.Printf("[DEBUG] cloudamqp::resource::alarm::create type is 'notice', getting existing notice alarm")
+		alarms, err := api.ReadAlarms(d.Get("instance_id").(int))
 
-	if err != nil {
-		return err
-	}
-	if data["id"] != nil {
-		d.SetId(data["id"].(string))
-		log.Printf("[DEBUG] cloudamqp::resource::alarm::create id set: %v", d.Id())
+		if err != nil {
+			return err
+		}
+
+		for _, alarm := range alarms {
+			if alarm["type"] == "notice" {
+				d.SetId(alarm["id"].(string))
```

Comment:
```suggestion
				d.SetId(strconv.FormatFloat(alarm["id"].(float64), 'f', 0, 64))
```
When I run this locally the plugin crashes due to wrong interface conversion. Contrary to data["id"] that is converted to a string in our client library, alarm["id"] is a float64 and cannot be asserted with string. For me to get this to work I had to convert it.